File: client/java/src/avt/ChatTextField.java
package avt;

import javax.microedition.lcdui.Graphics;
import main.Canvas;

public final class ChatTextField extends AvMain {
   public static ChatTextField instance;
   public TField tfChat;
   public static boolean isShow = false;
   public IChatable parentMyScreen;
   private long lastTimeChat;

   public final void commandTab(int var1, int var2) {
      switch (var1) {
         case 0:
            this.tfChat.setText("");
            isShow = false;
            this.tfChat.setFocus(true);
            return;
         case 1:
            long var3;
            if ((var3 = System.currentTimeMillis()) - this.lastTimeChat < 2000L) {
               return;
            } else if (this.parentMyScreen != null) {
               this.parentMyScreen.onChatFromMe(this.tfChat.getText());
               this.tfChat.setText("");
               isShow = false;
               this.tfChat.setFocus(true);
               this.lastTimeChat = var3;
            }
         default:
      }
   }

   public final void closeChat() {
      this.tfChat.setText("");
      isShow = false;
      this.tfChat.setFocus(true);
      if (OnScreen.isOngame && OptionScr.isVirTualKey) {
         OptionScr.isVirTualKey = false;
         OptionScr.gI().mapFocus[4] = 0;
         Canvas.instance.setSize();
      }

   }

   protected ChatTextField() {
      super.left = new Command(T.close, 0);
      super.center = new Command(T.chat, 1);
      this.tfChat = new TField();
      this.tfChat.e = false;
      this.tfChat.setFocus(true);
      this.init();
      this.tfChat.x = (Canvas.w - this.tfChat.width) / 2;
      this.tfChat.setMaxTextLenght(40);
      super.right = this.tfChat.getRightCmd();
   }

   public final void init() {
      this.tfChat.y = Canvas.hCan - Canvas.hTab - this.tfChat.height - 5;
      if (OnScreen.isOngame) {
         TField var10000 = this.tfChat;
         var10000.y -= 2 * AvMain.hd;
      }

      this.tfChat.width = Canvas.w - 10;
   }

   public final void keyPress(int var1) {
      if (isShow) {
         this.tfChat.keyPressed(var1);
      }

   }

   public static ChatTextField gI() {
      return instance == null ? (instance = new ChatTextField()) : instance;
   }

   public final void startChat(int var1, IChatable var2) {
      if (Canvas.currentFace == null) {
         this.tfChat.keyPressed(var1);
         if (!this.tfChat.getText().equals("")) {
            this.parentMyScreen = var2;
            isShow = true;
         }

         this.init();
      }

   }

   public final void updateKey() {
      this.tfChat.update();
      if (OnScreen.isOngame && Canvas.stypeInt != 0) {
         Canvas.paint.updateKeyOn(super.left, super.center, super.right);
      } else {
         super.updateKey();
      }

   }

   public final void paint(Graphics var1) {
      if (OnScreen.isOngame) {
         OnScreen.paintTitle(var1, super.left, super.center, super.right);
      } else {
         super.paint(var1);
      }

      this.tfChat.paint(var1);
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/ChatTextField.cs
using System;
using UnityEngine;

public class ChatTextField : AvMain
{
	private class IActionChat2 : IAction
	{
		public ChatTextField me;

		public IActionChat2(ChatTextField me)
		{
			this.me = me;
		}

		public void perform()
		{
			if (me != null && me.parentMyScreen != null)
			{
				string text = me.tfChat.getText();
				me.parentMyScreen.onChatFromMe(text);
				me.tfChat.setText(string.Empty);
				if (ipKeyboard.tk != null)
				{
					ipKeyboard.tk.text = string.Empty;
				}
				if (Screen.orientation == ScreenOrientation.Portrait)
				{
					isShow = false;
				}
				if (text.ToLower().Equals("cauca") && TouchScreenKeyboard.visible)
				{
					ipKeyboard.close();
					ipKeyboard.tk = null;
				}
			}
		}
	}

	public static ChatTextField instance;

	public TField tfChat;

	public static bool isShow;

	public IChatable parentMyScreen;

	private long lastTimeChat;

	private int chatButtonLight;

	protected ChatTextField()
	{
		center = new Command(T.chat, new IActionChat2(this));
		tfChat = new TField("chat", MapScr.instance, new IActionChat2(this));
		tfChat.setFocus(true);
		tfChat.showSubTextField = false;
		tfChat.autoScaleScreen = true;
		tfChat.setIputType(ipKeyboard.TEXT);
		init(Canvas.hCan);
		tfChat.x = 5 * AvMain.hd;
		tfChat.setMaxTextLenght(40);
		tfChat.action = new IActionChat2(this);
	}

	public void init(int hc)
	{
		tfChat.y = hc - tfChat.height - 7 * AvMain.hd;
		center = new Command(T.chat, new IActionChat2(this));
		if (onMainMenu.isOngame)
		{
			center.y = -200;
			tfChat.y = hc - tfChat.height - 7 * AvMain.hd - PaintPopup.hButtonSmall;
			tfChat.width = Canvas.w - 10 * AvMain.hd;
		}
		else
		{
			center.x = Canvas.w - MyScreen.wTab / 2 - 2 * AvMain.hd;
			center.y = tfChat.y + tfChat.height / 2 - PaintPopup.hButtonSmall / 2;
			tfChat.width = center.x - MyScreen.wTab / 2 - 10 * AvMain.hd;
		}
	}

	public void keyPressed(int keyCode)
	{
		if (isShow)
		{
			tfChat.keyPressed(keyCode);
		}
		if (tfChat.getText().Equals(string.Empty))
		{
			isShow = false;
		}
	}

	public static ChatTextField gI()
	{
		return (instance != null) ? instance : (instance = new ChatTextField());
	}

	public void startChat(int firstCharacter, IChatable parentMyScreen)
	{
		if (Canvas.currentFace == null && !ipKeyboard.tk.text.Equals(string.Empty))
		{
			tfChat.keyPressed(firstCharacter);
			this.parentMyScreen = parentMyScreen;
			isShow = true;
			tfChat.setFocusWithKb(true);
		}
	}

	public override void updateKey()
	{
		TField.currentTField = tfChat;
		if (onMainMenu.isOngame)
		{
			Canvas.paint.updateKeyOn(left, center, right);
		}
		else
		{
			base.updateKey();
		}
		if (chatButtonLight > 0)
		{
			chatButtonLight--;
		}
		tfChat.update();
	}

	public override void paint(MyGraphics g)
	{
		if (onMainMenu.isOngame)
		{
			Canvas.resetTrans(g);
			Canvas.paint.paintCmdBar(g, left, center, right);
		}
		else
		{
			base.paint(g);
		}
		tfChat.paint(g);
		g.setClip(0f, 0f, Canvas.w, Canvas.h);
		Canvas.resetTrans(g);
	}

	public void showTF()
	{
		if (!isShow)
		{
			tfChat.setFocus(true);
			try
			{
				parentMyScreen = (IChatable)Canvas.currentMyScreen;
				isShow = true;
				tfChat.parent = Canvas.currentMyScreen;
				tfChat.setFocusWithKb(true);
				Canvas.isPointerClick = false;
				Canvas.isPointerRelease = false;
			}
			catch (Exception)
			{
				try
				{
					parentMyScreen = MapScr.gI();
					isShow = true;
					tfChat.parent = Canvas.currentMyScreen;
					tfChat.setFocusWithKb(true);
					Canvas.isPointerClick = false;
					Canvas.isPointerRelease = false;
				}
				catch (Exception)
				{
				}
			}
		}
		Out.println("showTF: " + isShow);
	}

	public void showTF(string text)
	{
		if (!isShow)
		{
			tfChat.setFocus(true);
			try
			{
				parentMyScreen = (IChatable)Canvas.currentMyScreen;
				isShow = true;
				tfChat.parent = Canvas.currentMyScreen;
				tfChat.setText(text);
				tfChat.setFocusWithKb(true);
				Canvas.isPointerClick = false;
				Canvas.isPointerRelease = false;
			}
			catch (Exception)
			{
				try
				{
					parentMyScreen = MapScr.gI();
					isShow = true;
					tfChat.parent = Canvas.currentMyScreen;
					tfChat.setText(text);
					tfChat.setFocusWithKb(true);
					Canvas.isPointerClick = false;
					Canvas.isPointerRelease = false;
				}
				catch (Exception)
				{
				}
			}
		}
		Out.println("showTF: " + isShow);
	}
}
